packages/financial-templates-lib/src/proxy-transaction-handler/DSProxyManager.js

```diff
@@ -0,0 +1,148 @@
+  async initializeDSProxy() {
+    this.logger.debug({
+      at: "DSProxyManager",
+      message: "Initalizing...Looking for existing DSProxies or deploying a new one for the provided EOA",
+      dsProxyFactoryAddress: this.dsProxyFactoryAddress
+    });
+
+    const fromBlock = await getFromBlock(this.web3);
+    const events = await this.dsProxyFactory.getPastEvents("Created", {
+      fromBlock,
+      filter: { owner: this.account }
+    });
+
+    // The user already has a DSProxy deployed. Load it in from the events.
+    if (events.length > 0) {
+      this.dsProxyAddress = events[events.length - 1].returnValues.proxy;
+      this.dsProxy = new web3.eth.Contract(this.dsProxyAbi, this.dsProxyAddress);
+      this.logger.debug({
+        at: "DSProxyManager",
+        message: "DSProxy has been loaded in for the EOA",
+        dsProxyAddress: this.dsProxyAddress,
+        tx: events[events.length - 1].transactionHash,
+        account: this.account
+      });
+    }
+
+    // The user does not yet have a DSProxy. Create them one.
+    if (events.length == 0) {
+      this.logger.debug({
+        at: "DSProxyManager",
+        message: "No DSProxy found for EOA. Deploying new DSProxy",
+        account: this.account
+      });
+      const dsProxyCreateTx = await this.dsProxyFactory.methods.build().send({
+        from: this.account,
+        gasPrice: this.gasEstimator.getCurrentFastPrice()
+      });
+      this.dsProxyAddress = dsProxyCreateTx.events.Created.returnValues.proxy;
+      this.dsProxy = new web3.eth.Contract(this.dsProxyAbi, this.dsProxyAddress);
+      this.logger.info({
+        at: "DSProxyManager",
+        message: "DSProxy has been deployed for the EOA 🚀",
+        dsProxyAddress: this.dsProxyAddress,
+        tx: dsProxyCreateTx.transactionHash,
+        account: this.account
+      });
+    }
+  }
+  // Encode with `yourTruffleContractInstance.yourMethod(params1,param2).encodeABI()
+  async callFunctionOnExistingLibrary(libraryAddress, callData) {
+    assert(this.isAddress(libraryAddress), "Library address must be valid address");
+    assert(typeof callData === "string", "Call data must be a string");
+    this.logger.debug({
+      at: "DSProxyManager",
+      message: "Executing function on deployed library",
+      libraryAddress,
+      callData
+    });
+
+    const executeTransaction = await this.dsProxy.methods["execute(address,bytes)"](libraryAddress, callData).send({
+      from: this.account,
+      gasPrice: this.gasEstimator.getCurrentFastPrice()
+    });
+
+    this.logger.info({
+      at: "DSProxyManager",
+      message: "Executed function on deployed library 📸",
+      libraryAddress,
+      callData,
+      tx: executeTransaction.transactionHash
+    });
+    return executeTransaction;
+  }
+  // Extract call code using the `.abi` syntax on a truffle object or the `getABI(contractType,contractVersion)` from common.
+  async callFunctionOnNewlyDeployedLibrary(callCode, callData) {
```

Comment:
Is there a simple way (without a bunch of extra methods or code) to allow callers to simulate these transactions and grab the return value? aka just a .call rather than a .send()?

Reply:
ye, should be pretty easy to do I guess. we could make a separate method to do it and then just return that value there. WDYT?